NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -256,3 +415,517 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
+        else:
+            logger.info(f'electric field interpolation with lowfreq {interp_lowfreq/units.MHz} MHz and highfreq {interp_highfreq/units.MHz} MHz')
+            self.efield_interpolator = cr_pulse_interpolator.signal_interpolation_fourier.interp2d_signal(
+                self.obs_positions_vxB_vxvxB[:, 0],
+                self.obs_positions_vxB_vxvxB[:, 1],
+                self.electric_field_on_sky,
+                lowfreq=(interp_lowfreq-0.01)/units.MHz,
+                highfreq=(interp_highfreq+0.01)/units.MHz,
+                sampling_period=1/self.sampling_rate/units.s,  # interpolator module wants sampling period in seconds
+                phase_method="phasor",
+                radial_method='cubic',
+                upsample_factor=5,
+                coherency_cutoff_threshold=0.9,
+                ignore_cutoff_freq_in_timing=False,
+                verbose=False
+            )
+        return self.efield_interpolator
+
+    def initialize_fluence_interpolator(self, quantity=efp.signal_energy_fluence, debug=False):
+        """
+        initilized fluence interpolator object.
+
+        Parameters
+        ----------
+        quantity : electric field parameter
+            quantity to interpolate, e.g. efp.signal_energy_fluence
+            The quantity needs to be available as parameter in the electric field object!!! You might
+            need to run the electricFieldSignalReconstructor. The default is efp.signal_energy_fluence.
+        Returns
+        -------
+        fluence_interpolator : interpolator object
+        """
+        zz = [np.sum(efield[quantity]) for efield in self.sim_station.get_electric_fields()]  # the fluence is calculated per polarization, so we need to sum them up
+        self.fluence_interpolator_initilized = True
+
+        logger.info(f'fluence interpolation')
+        self.fluence_interpolator = cr_pulse_interpolator.interpolation_fourier.interp2d_fourier(
+            self.obs_positions_vxB_vxvxB[:, 0],
+            self.obs_positions_vxB_vxvxB[:, 1],
+            zz
+        )
+        if debug:
+            max_efield = []
+            for i in range(len(self.electric_field_on_sky[:,0,1])):
+                max_efield.append(np.max(np.abs(self.electric_field_on_sky[i,:,:])))
+            plt.scatter(self.obs_positions_vxB_vxvxB[:,0], self.obs_positions_vxB_vxvxB[:,1], c=max_efield, cmap='viridis', marker='o', edgecolors='k')
+            cbar = plt.colorbar()
+            cbar.set_label('max amplitude')
+            plt.xlabel('v x B [m]')
+            plt.ylabel('v x v x B [m]')
+            plt.show()
+            plt.close()
+
+        return self.fluence_interpolator
+
+    def plot_footprint_fluence(self, radius=300):
+        """
+        plots the interpolated values of the fluence in the shower plane
+
+        Parameters
+        ----------
+        radius : float
+            radius around shower core which should be plotted
+
+        save_file_path : str
```

Comment:
This is not a parameter of the function?

Reply:
a true, I changed it.